### PR TITLE
Magiclysm: Black Dragons and related gear

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Mods "Magiclysm: Black Dragons and related gear"```

Adds black dragon wyrmlings and young black dragons. they spawn rarely in swamps. Also adds a black dragons book that teaches some spellcraft, and various dragon armor recipes. the hide armor is based roughly on survivor armor, while the scale armor is roughly based on heavy survivor armor. the main benefits of dragon armor is the reduced weight and encumbrance, and the resistance related to the particular dragon it was made from.

### CODE REVIEW

File: data/mods/Magiclysm/monsters/dragon.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      "ACIDPROOF",
+      "PUSH_MON",
+      "FLIES"
+    ]
+  },
+  {
+    "id": "mon_dragon_black_young",
+    "type": "MONSTER",
+    "name": "young black dragon",
+    "description": "This black dragon appears to still be in the early stages of life.  Its eyes have just started to sink in its sockets, and its curved, segemented horns have just begun to darken at the tips.  You can just tell by looking at it that this creature is Evil to its very core.  Even though this dragon is not fully grown, it is the size of a full-grown bull.",
```
Comment (NickGrease) [inline data/mods/Magiclysm/monsters/dragon.json]:
```suggestion
    "description": "This black dragon appears to still be in the early stages of life.  Its eyes have just started to sink in its sockets, and its curved, segmented horns have just begun to darken at the tips.  You can tell just by looking at it, this creature is evil to its very core.  Even though this dragon is not fully grown, it is the size of a full-grown bull.",
```

File: data/mods/Magiclysm/items/black_dragon_items.json [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+  },
+  {
+    "id": "boots_black_dragon_scale",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": "pair of black dragonscale boots",
+    "name_plural": "pairs of black dragonscale boots",
+    "description": "Boots made of black dragonscale.  Very protective, and surprisingly light.",
+    "weight": 945,
+    "volume": 13,
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/black_dragon_items.json]:
```suggestion
    "volume": "3250 ml",
```

File: data/mods/Magiclysm/items/alchemy_items.json [hunk 3]
```diff
@@ -186,5 +186,20 @@
...
     "volume": 1
+  },
+  {
+    "id": "dragon_essence",
+    "type": "COMESTIBLE",
+    "symbol": "~",
+    "color": "light_red",
+    "name": "dragon essence",
+    "description": "This is the magical essence from a dragon, distilled into a concentrated form.",
+    "volume": 1,
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/alchemy_items.json]:
```suggestion
    "volume": "250 ml,
```

File: data/mods/Magiclysm/items/black_dragon_items.json [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+  },
+  {
+    "id": "boots_black_dragon_hide",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": "pair of black dragonhide boots",
+    "name_plural": "pairs of black dragonhide boots",
+    "description": "Boots made of black dragonhide.  Very protective, and surprisingly light.",
+    "weight": 655,
+    "volume": 13,
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/black_dragon_items.json]:
```suggestion
    "volume": "3250 ml",
```

File: data/mods/Magiclysm/items/black_dragon_items.json [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+    "flags": [ "VARSIZE", "WATERPROOF", "STURDY" ]
+  },
+  {
+    "id": "helmet_black_dragon_scale",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": "black dragonscale helmet",
+    "description": "A helmet made from black dragonscale, held together with black dragonhide.  It comes equipped with a full face visor.",
+    "weight": 856,
+    "volume": 10,
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/black_dragon_items.json]:
```suggestion
    "volume": "2500 ml",
```

File: data/mods/Magiclysm/items/black_dragon_items.json [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+    "flags": [ "VARSIZE", "WATERPROOF", "STURDY" ]
+  },
+  {
+    "id": "helmet_black_dragon_hide",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": "black dragonhide helmet",
+    "description": "A helmet made from black dragonhide.  It protects your head well, and doesn't cover your face.",
+    "weight": 585,
+    "volume": 10,
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/black_dragon_items.json]:
```suggestion
    "volume": "2500 ml",
```

File: data/mods/Magiclysm/items/black_dragon_items.json [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+  },
+  {
+    "id": "gauntlets_black_dragon_scale",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": "pair of black dragonscale gauntlets",
+    "name_plural": "pairs of black dragonscale gauntlets",
+    "description": "A pair of heavy-duty gauntlets made of black dragonscale that covers your hands.",
+    "weight": 380,
+    "volume": 4,
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/black_dragon_items.json]:
```suggestion
    "volume": "1 L",
```

File: data/mods/Magiclysm/items/black_dragon_items.json [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+  },
+  {
+    "id": "gloves_black_dragon_hide",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": "pair of black dragonhide gloves",
+    "name_plural": "pairs of black dragonhide gloves",
+    "description": "A pair of customized, Kevlar armored leather gloves, modified to be easy to wear while providing maximum protection under extreme conditions.",
+    "weight": 230,
+    "volume": 3,
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/black_dragon_items.json]:
```suggestion
    "volume": "750 ml",
```

File: data/mods/Magiclysm/items/recipe_books.json [hunk 9]
```diff
@@ -0,0 +1,21 @@
+[
+  {
+    "id": "black_dragons",
+    "type": "BOOK",
+    "name": "Black Dragons: Swamp Ruins",
+    "description": "This book details black dragons: The most evil of dragons, who live in ruins generally in swamps.  Immune to acid and breathe acid.  The book also details how to prepare its hide and craft various armors that make use of some of its acid resistance.",
+    "weight": 1000,
+    "volume": 3,
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/recipe_books.json]:
```suggestion
    "volume": "750 ml",
```

File: data/mods/Magiclysm/items/alchemy_items.json [hunk 10]
```diff
@@ -186,5 +186,20 @@
...
     "volume": 1
+  },
+  {
+    "id": "dragon_essence",
+    "type": "COMESTIBLE",
+    "symbol": "~",
+    "color": "light_red",
+    "name": "dragon essence",
+    "description": "This is the magical essence from a dragon, distilled into a concentrated form.",
+    "volume": "250 ml,
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/items/alchemy_items.json]: Whoops
```suggestion
    "volume": "250 ml",
```

**Comment by kevingranade** [comment]
```src/requirements.cpp:358 [void component::check_consistency(const string&) const] dragon_black_scale in dragon_scale is not a valid item template
```

**Pull request merged**